### DISCUSSION TITLE
architecture

### DISCUSSION THREAD

**Discussion by Alma593** [sarchlab/mgpusim#73] [Q&A] [answered]
Is there abstraction of grid, block or thread  in this simulator?

**Comment by syifan** [comment, marked as answer]
Yes. Please find the code definition here https://github.com/sarchlab/mgpusim/blob/v3/kernels/grid.go

In AMD terms, WorkGroup = block, Wavefront = Warp, Work-Item = Thread.

**Reply by Alma593** [reply]
thanks so much
